x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -0,0 +1,150 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import { EuiCallOut, EuiFilePicker, EuiFormRow, EuiSpacer, EuiText } from '@elastic/eui';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { isPlainObject } from 'lodash/fp';
+import type { IntegrationSettings } from '../../types';
+import * as i18n from './translations';
+
+const MaxLogsSampleRows = 10;
+
+/**
+ * Parse the logs sample file content (json or ndjson) and return the parsed logs sample
+ */
+const parseLogsContent = (
+  fileContent: string | undefined,
+  fileType: string
+): { error?: string; isTruncated?: boolean; logsSampleParsed?: string[] } => {
+  if (fileContent == null) {
+    return { error: 'Failed to read the logs sample file' };
+  }
+  let parsedContent;
+  try {
+    if (fileType === 'application/json') {
+      parsedContent = JSON.parse(fileContent);
+    } else if (fileType === 'application/x-ndjson') {
+      parsedContent = fileContent
+        .split('\n')
+        .filter((line) => line.trim() !== '')
+        .map((line) => JSON.parse(line));
+    }
+  } catch (_) {
+    return { error: `The logs sample file has not a valid ${fileType} format` };
+  }
+
+  if (!Array.isArray(parsedContent)) {
+    return { error: 'The logs sample file is not an array' };
+  }
+  if (parsedContent.length === 0) {
+    return { error: 'The logs sample file is empty' };
+  }
+
+  let isTruncated = false;
+  if (parsedContent.length > MaxLogsSampleRows) {
+    parsedContent = parsedContent.slice(0, MaxLogsSampleRows);
+    isTruncated = true;
+  }
+
+  if (parsedContent.some((log) => !isPlainObject(log))) {
+    return { error: 'The logs sample file contains non-object entries' };
+  }
+
+  const logsSampleParsed = parsedContent.map((log) => JSON.stringify(log));
+  return { isTruncated, logsSampleParsed };
+};
+
+interface SampleLogsInputProps {
+  integrationSettings: IntegrationSettings | undefined;
+  setIntegrationSettings: (param: IntegrationSettings) => void;
+}
+export const SampleLogsInput = React.memo<SampleLogsInputProps>(
+  ({ integrationSettings, setIntegrationSettings }) => {
+    const { notifications } = useKibana().services;
+    const [isParsing, setIsParsing] = useState(false);
+    const [sampleFileError, setSampleFileError] = useState<string>();
+
+    const onChangeLogsSample = useCallback(
+      (files: FileList | null) => {
+        const logsSampleFile = files?.[0];
+        if (logsSampleFile == null) {
+          setSampleFileError(undefined);
+          setIntegrationSettings({ ...integrationSettings, logsSampleParsed: undefined });
+          return;
+        }
+
+        const reader = new FileReader();
+        reader.onload = function (e) {
+          const fileContent = e.target?.result as string | undefined; // we use readAsText so this should be a string
```

Comment:
```suggestion
          const fileContent = e.target?.result ?? ''; // we use readAsText so this will be defined
```
I think this should work to avoid a typecast?

Reply:
Not really, the type of the `e.target.result` is `string | ArrayBuffer | null | undefined`, this cast is needed to tell TS that this won't be an _ArrayBuffer_.

We could implement a _typeof_ check and so on, but I think that would be even more confusing. Since we are explicitly calling `reader.readAsText` at line 107, we can safely cast it to `string` here.

If we were calling `reader.readAsArrayBuffer` ([docs](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsArrayBuffer)) we would receive an _ArrayBuffer_ as a result.

I will make the comment more explanatory. Is that okay?

Reply:
https://github.com/elastic/kibana/commit/f2fe6207796026eb8d3b935b3270be3a43f7c9c2#diff-fa374325098a7b72d48803bdaacc8ed2684655e97a80d2ac2ffa2014ee2e294cR83